### ISSUE TITLE
Playlist thumbnails sometimes are cached from other videos

### ISSUE THREAD

**Issue by stephendonner** [brave/brave-browser#37537]
## Description 

Playlist thumbnails sometimes are cached from other videos

## Steps to Reproduce

   1. install `1.67.2`
   2. launched Brave
   3. enabled `Playlist` via `brave://flags`
   4. queued 5 or so videos (didn't play any yet)
   5. played one
   6. looked at the thumbnails of the various videos

## Actual result:

<img width="1312" alt="Screen Shot 2024-04-11 at 5 56 38 PM" src="https://github.com/brave/brave-browser/assets/387249/d342c886-ce08-4885-a0a6-8fd00bb57f1c">

## Expected result:

Either the blurple `Brave` placeholder, or the actual video's thumbnail

## Reproduces how often: 

🤷 

## Brave version (brave://version info)

```
Brave	1.67.2 Chromium: 124.0.6367.29 (Official Build) nightly (x86_64) 
Revision	bcc8ab0f30ea7af989211434c17eb432f9375792
OS	macOS Version 11.7.10 (Build 20G1427)
```

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

cc @sangwoo108 @szilardszaloki @rebron @brave/qa-team

**Comment by sangwoo108** [comment]
Yeah, this is currently known issue. We're looking for a solution for this. cc. @szilardszaloki
